sa/model_test.go

```diff
@@ -35,7 +36,8 @@ func TestModelToRegistrationBadJSON(t *testing.T) {
 		Key: badJSON,
 	})
 	test.AssertError(t, err, "expected error from truncated reg model key")
-	badJSONErr, ok := err.(errBadJSON)
+	var badJSONErr errBadJSON
+	ok := errors.As(err, &badJSONErr)
```

Comment:
This can be combined with the next line:

`test.Assert(errors.As(err, &badJSONErr))`

Same elsewhere in this PR.

Reply:
Ended up settling on AssertEquals as it seemed cleanest.